FreeTubeApp/FreeTube
# List sorting & display settings

# List sorting & display settings




## Pull Request Type


## Related issue
#4096, [comment](https://github.com/FreeTubeApp/FreeTube/issues/1739#issuecomment-925846285)

## Description
- Implements radio group support for `ft-icon-button` dropdown
- Implements `ft-button` option for `ft-icon-button` base button appearance
- Implements three buttons in Theme Settings for fitting three of our features with new capabilities:
  1) Settings Sections,
     - Use default sort (default), sort A to Z, sort Z to A
   2) Side Nav subscription bubbles, and
   3) the `ft-profile-selector` dropdown:
     - Sort A to Z, sort Z to A
     - Display type: list or grid view
       - (grid-specific): Number of items per grid row
       - (grid-specific): Show or hide titles
   

## Video 

This video demos everything listed in the Testing section:

https://github.com/FreeTubeApp/FreeTube/assets/84899178/84b554c7-e7ed-4b73-b8e2-3b1b5f373701

Edit: After minor updates to settings placement and changing the settings section sort:
![Screenshot_20231026_144250](https://github.com/FreeTubeApp/FreeTube/assets/84899178/116322f0-d8bd-42ed-b220-df76d9a8bbeb)

## Testing 
- Test radio group functionality of the `ft-icon-button` dropdown:
  - Working up/left/down/right arrow navigation
  -  `Tab` / `Shift` + `Tab` brings you out of the radio group to the next / previous tabbable element
  - `Spacebar` checks the focused checkbox
- Check that the expanded side nav, settings section, and profile selector dropdown behave normally with `settings.js` defaults
- Check that side nav subscriptions look good with every combination of `grid`, `itemsPerGridRow`, and `showTitles` and respects the selected sorting order
- Check that the profile selector looks good with every combination of `grid`, `itemsPerGridRow`, and `showTitles` and respects the selected sorting order
- Check that the settings sections respect the selected sorting order

## Desktop

- **OS:** OpenSUSE Tumbleweed
- **OS Version:** 2023xxxx
- **FreeTube version:** 0.19.1

## Additional context
Remaining related discussions: #1739, #987, #800, parts of #4096, #2871

The new radio group support in the `ft-icon-button` dropdown was just created as a helpful tool for bringing about what I was envisioning, but I can see it coming in handy again for creating related settings in a space-compressed and low-cognitive-complexity way. See discussion under #2295.

Among other reasons, I also created this when pondering the design of this issue that I intend to work on: #1615.

Please let me know how you feel about the default grid values chosen in `settings.js` (in scope for this PR) or if we would consider changing any of the default display values / sorting methods (broader discussion out of scope for this PR). I'm personally going to be using the grid + 3 per grid row + show titles for subscriptions and grid + 4 per grid row + show titles for the profile selector during my personal use of FreeTube.
